src/main/java/frc/team2412/robot/subsystems/ClimbSubsystem.java

```diff
@@ -0,0 +1,121 @@
+package frc.team2412.robot.subsystems;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj.CAN;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.team2412.robot.Hardware;
+import frc.team2412.robot.Robot;
+
+import com.revrobotics.CANSparkFlex;
+import com.revrobotics.SparkAbsoluteEncoder;
+import com.revrobotics.SparkPIDController;
+import com.revrobotics.CANSparkLowLevel.MotorType;
+import com.revrobotics.SparkAbsoluteEncoder.Type;
+import com.revrobotics.CANSparkBase.ControlType;
+import com.revrobotics.CANSparkBase.IdleMode;
+
+public class ClimbSubsystem extends SubsystemBase{
+    //public or private?
+    //CONSTANT VARIABLES
+        //speeds (fix later)
+    private final double EXTEND_SPEED = 0.1;
+    private final double RETRACT_SPEED = -0.1;
+    //private final double MOTOR_SPEED = 0.15;
+        //PID extension values
+        //PID retraction values
+        //height of chain in encoder??? units
+
+    //  INSTANCE VARIABLES
+    private final CANSparkFlex motorLeft;
+    private final CANSparkFlex motorRight;
+    private final SparkAbsoluteEncoder encoderLeft;
+    private final SparkAbsoluteEncoder encoderRight;
```

Comment:
Should delete these, as the absolute encoder is only useful for measuring less than a single turn. Assuming climb rotates more than one turn, the `RelativeEncoder` returned from `motorLeft.getEncoder()` is what will need to be used.